### PR TITLE
Polling station status behavior

### PR DESCRIPTION
**Pull request by cikzh**
- Adds an alert when the users enters a valid polling station number, but it's in the "Definitief" state
- Filters polling stations from the list with a "Definitief" state
- Throws an error and send the user to the "Er is iets mis" page, in case the user tries to reach a finalised polling station through the URL bar or browser history

@jorisleker When the user tries to select a finalised polling station, it will get an alert, but at the same time the polling station message will appear green. Perhaps we want the polling station message to be a different color? Yellow or red? What do you think?

### CODE REVIEW

File: frontend/app/component/form/data_entry/polling_station_choice/PollingStationChoiceForm.tsx [hunk 1]
```diff
@@ -38,17 +43,24 @@ export function PollingStationChoiceForm({ anotherEntry }: PollingStationChoiceF
...
-      setShowAlert(true);
+      setAlert(INVALID_POLLING_STATION_ALERT);
       return;
     }
 
     const parsedStationNumber = parsePollingStationNumber(pollingStationNumber);
     const pollingStation = pollingStations.find((pollingStation) => pollingStation.number === parsedStationNumber);
+    const pollingStationStatus = electionStatuses.statuses.find((status) => status.id === pollingStation?.id);
+
+    if (pollingStationStatus?.status === "definitive") {
```
Comment (Lionqueen94) [inline frontend/app/component/form/data_entry/polling_station_choice/PollingStationChoiceForm.tsx:54]: It looks a bit weird to use pollingStationStatus.status, maybe we can rename the const?

File: frontend/app/routes.test.tsx [hunk 2]
```diff
@@ -90,4 +90,22 @@ describe("routes", () => {
...
+    await router.navigate("/elections/1/data-entry/2");
+    expect(router.state.location.pathname).toEqual("/elections/1/data-entry/2");
+
+    // NOTE: We're not using the wrapped render function here,
+    // since we want control over our own memory router.
+    rtlRender(
+      <ApiProvider>
+        <RouterProvider router={router} />
+      </ApiProvider>,
+    );
```
Comment (praseodym) [inline frontend/app/routes.test.tsx:107]: We can simplify this after my changes in https://github.com/kiesraad/abacus/pull/233/commits/cc2afed8e602cdca510e1f48ed850295d440f806:
```typescript
rtlRender(<Providers router={router} />);
```
Or to replace all invocations in this file:
```typescript
// top of file
const render = () => rtlRender(<Providers router={router} />);

// in the test
render();
```
and replace all duplicate code blocks.

**Review by praseodym — COMMENTED** [review]
Two minor comments, otherwise it looks good!

Also, maybe not for this PR/issue, I think the design of the polling station choice page can be improved for when all inputs have been completed. See this screenshot:
localhost_3000_elections_1_data-entry (2)
The green alert isn't even meant for the 'invoerder' role, so we'll be removing this in a later iteration. @jorisleker any ideas on what we should show in that case?

File: frontend/app/routes.test.tsx (hunk 2 above)
Comment (cikzh) [reply frontend/app/routes.test.tsx]: [72a87fe](https://github.com/kiesraad/abacus/pull/367/commits/72a87fee88145db3182e255116d1662337f4a3f0)

File: frontend/app/component/form/data_entry/polling_station_choice/PollingStationChoiceForm.tsx (hunk 1 above)
Comment (cikzh) [reply frontend/app/component/form/data_entry/polling_station_choice/PollingStationChoiceForm.tsx]: [a932acc](https://github.com/kiesraad/abacus/pull/367/commits/a932accef1b712d26fa3d2a86a502af6202dc2fd)

File: frontend/app/component/form/data_entry/polling_station_choice/PollingStationsList.tsx [hunk 3]
```diff
@@ -28,6 +28,10 @@ export function PollingStationsList({ pollingStations }: PollingStationsListProp
       <tbody>
         {pollingStations.map((pollingStation: PollingStation) => {
           const pollingStationStatus = electionStatus.statuses.find((status) => status.id === pollingStation.id);
+          if (pollingStationStatus?.status === "definitive") {
```
Comment (Lionqueen94) [inline frontend/app/component/form/data_entry/polling_station_choice/PollingStationsList.tsx:31]:
```suggestion
          const pollingStationStatusEntry = electionStatus.statuses.find((status) => status.id === pollingStation.id);
          if (pollingStationStatusEntry?.status === "definitive") {
```

**Review by praseodym — DISMISSED** [review]

**Review by Lionqueen94 — APPROVED** [review]

**Comment by praseodym** [comment]
> WDYT? Shall I create a new ticket to implement these?

Yes please!

**Comment by jorisleker** [comment]
✅ https://github.com/kiesraad/abacus/issues/388

**Pull request merged**